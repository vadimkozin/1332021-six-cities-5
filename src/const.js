export const CITIES = {
  Paris: [48.856614, 2.3522219],
  Cologne: [50.937531, 6.960278600000038],
  Brussels: [50.8503396, 4.3517103],
  Amsterdam: [52.3702157, 4.8951679],
  Hamburg: [53.5510846, 9.99368179999999],
  Dusseldorf: [51.2277411, 6.773455600000034],
};
export const CITIES_NAMES = Object.keys(CITIES);

export const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

export const RATING_STARS_MAX = 5;

export const OFFER_PICTURE_MAX = 6;

export const REVIEW_OUTPUT_MAX = 10;

export const MAP_ZOOM_DEFAULT = 12;

export const HOTELS_NEARBY_MAX = 3;

export const CommentSettings = {
  TEXT_LENGTH_MIN: 50,
  TEXT_LENGTH_MAX: 300,
};

export const TypesHousing = {
  APARTMENT: {name: `apartment`, view: `Apartment`},
  ROOM: {name: `room`, view: `Private Room`},
  HOUSE: {name: `house`, view: `House`},
  HOTEL: {name: `hotel`, view: `Hotel`}
};

export const OfferCardType = {
  CITY_PLACE: 1,
  NEAR_PLACE: 2,
};

export const MapType = {
  VERTICAL: 1,
  HORIZONTAL: 2,
};

export const SortingType = {
  POPULAR: `popular`, // в порядке, полученном с сервера
  LOW_TO_HIGH: `to-high`, // От дешёвых к дорогим.
  HIGH_TO_LOW: `to-low`, // От дорогих к дешёвым.
  TOP_RATED_FIRST: `top-rated` // От высокого рейтинга к низкому
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER_ID: `/offer/:id`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
  OFFER_ID: `/hotels/`,
};

export const StatusType = {
  IDLE: `idle`,
  LOADING: `loading`,
  SUCCESS: `succes`,
  FAILED: `failed`,
};
