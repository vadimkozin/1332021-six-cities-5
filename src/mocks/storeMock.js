import configureStore from 'redux-mock-store';
import {SortingType, CITIES_NAMES, StatusType, AuthorizationStatus} from '../const';
import {offerAdapter, commentAdapter} from '../store/adapters';

import hotelJson from './json/hotel';
import hotelsNearbyJson from './json/hotels-nearby';
import offersJson from './json/offers';
import commentsJson from './json/comments';

const hotel = offerAdapter.adaptToClientOffer(hotelJson);
const hotelsNearby = offerAdapter.adaptToClient(hotelsNearbyJson);
const offers = offerAdapter.adaptToClient(offersJson);
const comments = commentAdapter.adaptToClient(commentsJson);

const initCity = CITIES_NAMES[0];

const Data = {
  SORT: `sort`,
  CITY: `city`,
  OFFERS: `offersGist`,
  HOTEL: `hotelGist`,
  HOTELS_NEARBY: `hotelsNearbyGist`,
  COMMENTS: `commentsGist`,
};

const basicState = {
  data: null,
  status: StatusType.IDLE,
  error: ``,
};

const makeInit = (basic = basicState) => (data = {}) => Object.assign({}, basic, {data});

const init = makeInit();

const storeConfig = configureStore();

const storeMock = storeConfig({
  data: {
    cities: CITIES_NAMES,
  },
  process: {
    [Data.CITY]: initCity,
    [Data.SORT]: SortingType.POPULAR,
    [Data.OFFERS]: init(offers),
    [Data.HOTEL]: init(hotel),
    [Data.HOTELS_NEARBY]: init(hotelsNearby),
    [Data.COMMENTS]: init(comments),
  },
  user: {
    id: 1,
    name: `qwerty`,
    email: `qwerty@mail.ru`,
    isPro: false,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
});

export {storeMock};
