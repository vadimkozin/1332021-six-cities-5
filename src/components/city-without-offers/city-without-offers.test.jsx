import React from 'react';
import {Provider} from 'react-redux';
import {render} from 'enzyme';
import storeMock from '../../mocks/storeMock';
import CityWithoutOffers from './city-without-offers';
import {CITIES_NAMES} from '../../const';

describe(`CityWithoutOffers`, () => {
  it(`should render CityWithoutOffers component`, () => {
    const component = render(
        <Provider store={storeMock}>
          <CityWithoutOffers city={CITIES_NAMES[0]}/>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
