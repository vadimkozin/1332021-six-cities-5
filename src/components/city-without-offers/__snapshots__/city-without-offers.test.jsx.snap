// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWithoutOffers should render CityWithoutOffers component 1`] = `
<section
  class="cities__no-places"
>
  <div
    class="cities__status-wrapper tabs__content"
  >
    <b
      class="cities__status"
    >
      No places to stay available
    </b>
    <p
      class="cities__status-description"
    >
      We could not find any property available at the moment in Paris
    </p>
  </div>
</section>
`;
